include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -160,6 +175,25 @@ MGConstrainedDoFs::initialize(const DoFHandler<dim,spacedim> &dof,
 }
 
 
+template <int dim, int spacedim>
+inline
+void
+MGConstrainedDoFs::set_zero_boundary_dofs(const DoFHandler<dim,spacedim> &dof,
+                                          const std::set<types::boundary_id> &boundary_indicators,
+                                          const ComponentMask &component_mask)
+{
+  // allocate an IndexSet for each global level. Contents will be
+  // overwritten inside make_boundary_list.
+  const unsigned int n_levels = dof.get_triangulation().n_global_levels();
+  boundary_indices.resize(n_levels);
```

Comment:
> |boundary_indices| is cleared in |initialize| and |clear|. Since this is a
> private variable there is no possibility that it has a size different from
> zero or n_levels.

Then maybe asserting that this is so would be the right thing to do. It also 
makes it clearer for readers like me that nothing else can happen.